src/App.tsx

```diff
@@ -47,13 +47,50 @@ const useAppStyles = makeStyles(theme => {
 const App = (): JSX.Element => {
   const classes = useAppStyles(theme);
 
-  // This Hook is used to keep track of whether to render the table or CommitmentDetails
-  // const [displayDetails, setDisplayDetails] = React.useState(false);
   // This Hook is used to keep track of which CommitmentDetails should be displayed
   const [displayId, setDisplayId] = React.useState(
     window.location.hash.slice(1) || ''
   );
 
+  // This is intended to be used for more generalized purposes
+  const [history, setHistory] = React.useState([] as string[]); // May not need to be a hook
+
+  const isInDoc = useRef(true);
+
+  // Intended for future functionality
+  const goBack = () => {
+    window.location.hash = history[history.length - 1];
+    setHistory(history.slice(0, -1)); // Should effectively be an inverted pop
+  };
+
+  useEffect(() => {
+    // The purpose of these event listeners is to determine the difference between the on-page and browser back buttons
+    window.addEventListener('mouseenter', () => {
+      isInDoc.current = true;
+    });
+    window.addEventListener('mouseleave', () => {
+      isInDoc.current = false;
+    });
+    window.addEventListener('hashchange', () => {
+      // If the in-doc back-button is used
+      if (isInDoc.current) {
+        // If the App is displaying table
+        if (window.location.hash === '') {
+          // Nothing needs to happen
+          setDisplayId('');
+        }
+      } else {
+        // If there is no browser history
+        if (window.location.hash !== '#undefined') {
+          setDisplayId('');
+          window.location.hash = '';
+        } else {
+          goBack();
+        }
+      }
+    });
+  });
```

Comment:
This is the code that matters most.  You don't really need to know if the back button is used.  You just need to know if the hash change was caused by a `back` click or caused by a user clicking a row in the table (this is the wonky part).

One approach might be to listen for hash changes, but disregard the hash change if you clicked a button in your app right before hand.  This would mean creating a function for setting the hash instead of saying `window.location.hash = <value>`.  Something like:

````
// somewhere outside the component:
let shouldDisregardNextHashChange = false;
...
const Component = () => {
  React.useEffect(() => {
    window.onhashchange = (ev: HashChangeEvent) => {
      if (shouldDisregardNextHashChange) {
        shouldDisregardNextHashChange = false;
        return;
      }

      const newHash = ev.newURL.slice(ev.newURL.indexOf('#'));
      if (newHash === '/' ) {
        setDisplayId('');
      } else {
        setDisplayId(newHash.slice(1));
      }
    };
  });

  const changeHash = (newHash: string) => {
    shouldDisregardNextHashChange = true;
    window.location.hash = newHash;
  };
  ...
}
````

Then, any time you were manually changing the hash, instead call `changeHash`